### PR TITLE
Adds PulpFileReplicator

### PR DESCRIPTION
**Pull request by dkliban**
closes: #909

### CODE REVIEW

File: CHANGES/909.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Adds ability to replicate File distributions/repositories from an upstream Pulp.
```
Comment (lubosmj) [inline CHANGES/909.feature:1]:
```suggestion
Added the ability to replicate File distributions/repositories from an upstream Pulp instance.
```

Comment (dkliban) [reply CHANGES/909.feature]: Thanks for the suggestion! I updated the changelog entry.

File: pulp_file/app/replica.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    remote_model_cls = FileRemote
+    repository_model_cls = FileRepository
+    distribution_model_cls = FileDistribution
+    distribution_serializer_name = "FileDistributionSerializer"
+    repository_serializer_name = "FileRepositorySerializer"
+    remote_serializer_name = "FileRemoteSerializer"
+    sync_task = file_synchronize
+
+    def url(self, upstream_distribution):
+        # Check if a distribution is repository or publication based
```
Comment (ipanova) [inline pulp_file/app/replica.py:27]: what about distribution which is repository_version based?

File: pulp_file/app/replica.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                self.pulp_ctx, upstream_distribution["publication"]
+            ).entity["manifest"]
+        else:
+            # This distribution doesn't serve any content
+            return None
+
+        return f"{upstream_distribution['base_url']}{manifest}"
+
+    def repository_extra_fields(self, remote):
+        return dict(manifest=remote.url.split("/")[-1], autopublish=True)
```
Comment (mdellweg) [inline pulp_file/app/replica.py:43]: Just a matter of style...
```suggestion
        return {"manifest": remote.url.split("/")[-1], "autopublish": True}
```

File: pulp_file/app/replica.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    def sync_params(self, repository, remote):
+        return dict(
+            remote_pk=str(remote.pk),
+            repository_pk=str(repository.pk),
+            mirror=True,
+        )
```
Comment (mdellweg) [inline pulp_file/app/replica.py:50]: Same:
```suggestion
    def sync_params(self, repository, remote):
        return {
            "remote_pk": str(remote.pk),
            "repository_pk": str(repository.pk),
            "mirror": True,
        }
```

File: pulp_file/app/replica.py (hunk 2 above)
Comment (ipanova) [reply pulp_file/app/replica.py:27]: Thanks for confirmation!

**Review by ipanova — APPROVED** [review]

**Pull request merged**